osu.Game/Screens/Menu/IntroTriangles.cs

```diff
@@ -72,9 +72,16 @@ protected override void LogoArriving(OsuLogo logo, bool resuming)
                     RelativeSizeAxes = Axes.Both,
                     Clock = decoupledClock,
                     LoadMenu = LoadMenu
-                }, t =>
+                }, _ =>
                 {
-                    AddInternal(t);
+                    AddInternal(intro);
+
+                    // There is a chance that the intro timed out before being displayed, and this scheduled callback could
+                    // happen during the outro rather than intro. In such a scenario the game may already be in a disposing state
+                    // which will trigger errors during attempted audio playback.
+                    if (DidLoadMenu)
+                        return;
```

Comment:
But this callback is scheduled to the intro, so it must not be disposed yet... https://github.com/ppy/osu-framework/blob/639a843f45f2c7344d4bf061deec6f2439b397c0/osu.Framework/Graphics/Containers/CompositeDrawable.cs#L195-L209

... And `DecouplableInterpolatingFramedClock.Start()` call does the disposal check synchronously.

Which means (and is indicated by Sentry) that all of this code is running on the update thread. This sounds like something that will cause endless problems for us, and I don't know what the cause is...

Reply:
I might need to reword the comment since it doesn't seem to be conveying the underlying issue: the *track* is disposed, because during gameplay the intro track has since been switched away from (and disposed by `MusicController`).

During the outro, the intro is visible for a second or two ("see you next time"), running its scheduler and attempting to start playing back the now-disposed track.

Reply:
To clarify a bit further, `Game`'s disposal will dispose children before the `AudioManager` (and relevant `TrackStore`s):
https://github.com/ppy/osu-framework/blob/639a843f45f2c7344d4bf061deec6f2439b397c0/osu.Framework/Game.cs#L432-L451

If this isn't happening in the correct order for some reason, then I'd hope that should be fixed framework-side.

Reply:
> during gameplay the intro track has since been switched away from (and disposed by MusicController).

Oh. I thought the failing thing in this case was `welcome`, but it's actually `Track`... Definitely need to mention something in the comment here. Still not sure how to feel about this change but it makes a bit more sense, I guess...

Reply:
Have reworded the comment.